### PR TITLE
must-gather: collect some of the odf-operator and storagesystem relat…

### PR DESCRIPTION
**Pull request by yati1998**
This commit collects log for the following:
* oc get  storagesystem -n openshift-storage

Signed-off-by: yati1998 <ypadia@redhat.com>

### CODE REVIEW

File: must-gather/collection-scripts/gather_namespaced_resources [hunk 1]
```diff
@@ -114,3 +113,22 @@ echo "collecting dump of oc get obc all namespaces" | tee -a  "${BASE_COLLECTION
...
+LOG_DIR=${BASE_COLLECTION_PATH}/namespaces/logs/${INSTALL_NAMESPACE}
+mkdir -p "${LOG_DIR}"
+{ timeout 120 oc -n "${INSTALL_NAMESPACE}" logs --all-containers "${operator_pod}" &> "${LOG_DIR}"/"${operator_pod}".log; } >> "${BASE_COLLECTION_PATH}"/gather-debug.log 2>&1
+
+# For oc logs odf-console pods
+echo "collecting oc command odf-console" | tee -a "${BASE_COLLECTION_PATH}/gather-debug.log"
+console_pod=$(oc get pods -n openshift-storage | grep odf-console | awk '{print $1}')
+LOG_DIR=${BASE_COLLECTION_PATH}/namespaces/logs/${INSTALL_NAMESPACE}
+mkdir -p "${LOG_DIR}"
+{ timeout 120 oc -n "${INSTALL_NAMESPACE}" logs --all-containers "${console_pod}" &> "${LOG_DIR}"/"${console_pod}".log; } >> "${BASE_COLLECTION_PATH}"/gather-debug.log 2>&1
```
Comment (jarrpa) [inline must-gather/collection-scripts/gather_namespaced_resources:134]: Is there no other place where we collect Pod logs? All must-gather output I've seen has included the logs of all Pods in the namespace, why is this required here?

**Review by jarrpa — CHANGES_REQUESTED** [review]

**Comment by jarrpa** [comment]
> @iamniting @jarrpa @umangachapagain Anything else we need to add here?

There sure is! :grin:

File: must-gather/collection-scripts/gather_namespaced_resources [hunk 2]
```diff
@@ -99,7 +99,6 @@ for resource in "${resources[@]}"; do
     { oc adm --dest-dir="${BASE_COLLECTION_PATH}/namespaces/all/" inspect "${resource}" --all-namespaces --"${SINCE_TIME}"; } >> "${BASE_COLLECTION_PATH}/gather-debug.log" 2>&1
 done
 
-
```
[the remark this answers is not in the document]
Comment (yati1998) [reply must-gather/collection-scripts/gather_namespaced_resources:102]: okay.

File: must-gather/collection-scripts/gather_namespaced_resources [hunk 3]
```diff
@@ -114,3 +113,22 @@ echo "collecting dump of oc get obc all namespaces" | tee -a  "${BASE_COLLECTION
 echo "collecting dump of oc get volumereplication all namespaces" | tee -a  "${BASE_COLLECTION_PATH}/gather-debug.log"
 { oc get volumereplication --all-namespaces; } >> "${BASE_COLLECTION_PATH}/namespaces/all/vr_all_namespaces"
 { oc adm --dest-dir="${BASE_COLLECTION_PATH}/namespaces/all/" inspect volumereplication --all-namespaces --"${SINCE_TIME}"; } >> "${BASE_COLLECTION_PATH}/gather-debug.log" 2>&1
+
+# For StorageSystem of all namespaces
+echo "collecting dump of oc get storagesystem all namespaces" | tee -a  "${BASE_COLLECTION_PATH}/gather-debug.log"
+{ oc get storagesystem --all-namespaces; } >> "${BASE_COLLECTION_PATH}/namespaces/all/ss_all_namespaces"
+{ oc adm --dest-dir="${BASE_COLLECTION_PATH}/namespaces/all/" inspect storagesystem --all-namespaces --"${SINCE_TIME}"; } >> "${BASE_COLLECTION_PATH}/gather-debug.log" 2>&1
```
[the remark this answers is not in the document]
Comment (yati1998) [reply must-gather/collection-scripts/gather_namespaced_resources:120]: The above list runs the get commands only for openshift namespace. But we need this for all nampespace.

File: must-gather/collection-scripts/gather_namespaced_resources (hunk 1 above)
Comment (yati1998) [reply must-gather/collection-scripts/gather_namespaced_resources:134]: No, for namespace we don't collect pod logs elsewhere. We have it in ceph and nooba for their resources.

**Comment by yati1998** [comment]
> We need `oc get storagesystem --all-namespaces -oyaml` and optionally `oc describe storagesystem --all-namespaces `. Current command only gives us a list.

@umangachapagain Do we need to collect logs for both oc describe and get commands? or oc describe works?

File: must-gather/collection-scripts/gather_namespaced_resources (hunk 2 above)
[the remark this answers is not in the document]
Comment (yati1998) [reply must-gather/collection-scripts/gather_namespaced_resources:102]: There is no need for 2 blank spaces. Hence, removed it.

**Comment by yati1998** [comment]
> We need `oc get storagesystem --all-namespaces -oyaml` and optionally `oc describe storagesystem --all-namespaces `. Current command only gives us a list.

done. please review it.

**Comment by yati1998** [comment]
> > @iamniting @jarrpa @umangachapagain Anything else we need to add here?
> 
> There sure is!

@jarrpa please do let me know what else is required.

File: must-gather/collection-scripts/gather_namespaced_resources [hunk 4]
```diff
@@ -114,3 +113,23 @@ echo "collecting dump of oc get obc all namespaces" | tee -a  "${BASE_COLLECTION
 echo "collecting dump of oc get volumereplication all namespaces" | tee -a  "${BASE_COLLECTION_PATH}/gather-debug.log"
 { oc get volumereplication --all-namespaces; } >> "${BASE_COLLECTION_PATH}/namespaces/all/vr_all_namespaces"
 { oc adm --dest-dir="${BASE_COLLECTION_PATH}/namespaces/all/" inspect volumereplication --all-namespaces --"${SINCE_TIME}"; } >> "${BASE_COLLECTION_PATH}/gather-debug.log" 2>&1
+
+# For StorageSystem of all namespaces
+echo "collecting dump of oc get storagesystem all namespaces" | tee -a  "${BASE_COLLECTION_PATH}/gather-debug.log"
+{ oc get storagesystem --all-namespaces; } >> "${BASE_COLLECTION_PATH}/namespaces/all/ss_all_namespaces"
+{ oc describe storagesystem --all-namespaces; } >> "${BASE_COLLECTION_PATH}/namespaces/all/ss_all_namespaces"
+{ oc get storageclusters --all-namespaces -o yaml; } >> "${BASE_COLLECTION_PATH}/namespaces/all/ss_all_namespaces.yaml"
```
Comment (umangachapagain) [inline must-gather/collection-scripts/gather_namespaced_resources:121]: Aren't we already collecting this?
It should probably be a separate block starting with `echo "collecting dump of storageclusters across all namespaces"`

File: must-gather/collection-scripts/gather_namespaced_resources [hunk 5]
```diff
@@ -114,3 +113,23 @@ echo "collecting dump of oc get obc all namespaces" | tee -a  "${BASE_COLLECTION
 echo "collecting dump of oc get volumereplication all namespaces" | tee -a  "${BASE_COLLECTION_PATH}/gather-debug.log"
 { oc get volumereplication --all-namespaces; } >> "${BASE_COLLECTION_PATH}/namespaces/all/vr_all_namespaces"
 { oc adm --dest-dir="${BASE_COLLECTION_PATH}/namespaces/all/" inspect volumereplication --all-namespaces --"${SINCE_TIME}"; } >> "${BASE_COLLECTION_PATH}/gather-debug.log" 2>&1
+
+# For StorageSystem of all namespaces
+echo "collecting dump of oc get storagesystem all namespaces" | tee -a  "${BASE_COLLECTION_PATH}/gather-debug.log"
```
Comment (umangachapagain) [inline must-gather/collection-scripts/gather_namespaced_resources:118]:
```suggestion
echo "collecting dump of storagesystems across all namespaces" | tee -a  "${BASE_COLLECTION_PATH}/gather-debug.log"
```

File: must-gather/collection-scripts/gather_namespaced_resources [hunk 6]
```diff
@@ -114,3 +113,23 @@ echo "collecting dump of oc get obc all namespaces" | tee -a  "${BASE_COLLECTION
...
+{ oc get storagesystem --all-namespaces; } >> "${BASE_COLLECTION_PATH}/namespaces/all/ss_all_namespaces"
+{ oc describe storagesystem --all-namespaces; } >> "${BASE_COLLECTION_PATH}/namespaces/all/ss_all_namespaces"
```
Comment (umangachapagain) [inline must-gather/collection-scripts/gather_namespaced_resources:120]:
```suggestion
{ oc get storagesystem --all-namespaces; } >> "${BASE_COLLECTION_PATH}/namespaces/all/storagesystem_all_ns"
{ oc describe storagesystem --all-namespaces; } >> "${BASE_COLLECTION_PATH}/namespaces/all/storagesystem_describe_all_ns"
```

File: must-gather/collection-scripts/gather_namespaced_resources (hunk 4 above)
Comment (umangachapagain) [reply must-gather/collection-scripts/gather_namespaced_resources:121]:
```suggestion
{ oc get storageclusters --all-namespaces -o yaml; } >> "${BASE_COLLECTION_PATH}/namespaces/all/storagesystem_yaml_all_ns"
```

**Review by umangachapagain — CHANGES_REQUESTED** [review]

File: must-gather/collection-scripts/gather_namespaced_resources (hunk 2 above)
[the remark this answers is not in the document]
Comment (jarrpa) [reply must-gather/collection-scripts/gather_namespaced_resources:102]: And it is also irrelevant to this PR. Remove it. At best it can be included as part of a larger commit to refactor and clean up the file.

File: must-gather/collection-scripts/gather_namespaced_resources (hunk 1 above)
Comment (jarrpa) [reply must-gather/collection-scripts/gather_namespaced_resources:135]: This statement seems absolutely wrong. Unless I'm missing something, this looks to me like we collect the logs of all Pods in the openshift-storage Namespace for every must-gather run: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_ocs-operator/1224/pull-ci-openshift-ocs-operator-master-ocs-operator-bundle-e2e-aws/1431303805407858688/artifacts/ocs-operator-bundle-e2e-aws/ocs-must-gather/artifacts/ocs-must-gather/registry-build02-ci-openshift-org-ci-op-s10lq9fp-stable-sha256-2cabcf61a08025e75eba8b82626357ddb48a44bf609f7d1483242ad5d0b1f5f0/namespaces/openshift-storage/pods/

I don't see anywhere in our scripts where we explicitly do this, so maybe it's part of the `oc adm inspect` of the namespace? In any case there's no reason to isolate these two specific Pods when we just want all logs in the namespace anyway.

File: must-gather/collection-scripts/gather_namespaced_resources [hunk 7]
```diff
@@ -114,3 +113,23 @@ echo "collecting dump of oc get obc all namespaces" | tee -a  "${BASE_COLLECTION
 echo "collecting dump of oc get volumereplication all namespaces" | tee -a  "${BASE_COLLECTION_PATH}/gather-debug.log"
 { oc get volumereplication --all-namespaces; } >> "${BASE_COLLECTION_PATH}/namespaces/all/vr_all_namespaces"
 { oc adm --dest-dir="${BASE_COLLECTION_PATH}/namespaces/all/" inspect volumereplication --all-namespaces --"${SINCE_TIME}"; } >> "${BASE_COLLECTION_PATH}/gather-debug.log" 2>&1
+
+# For StorageSystem of all namespaces
+echo "collecting dump of storagesystems across all namespaces" | tee -a  "${BASE_COLLECTION_PATH}/gather-debug.log"
+{ oc get storagesystem --all-namespaces; } >> "${BASE_COLLECTION_PATH}/namespaces/all/storagesystem_all_ns"
+{ oc describe storagesystem --all-namespaces; } >> "${BASE_COLLECTION_PATH}/namespaces/all/storagesystem_describe_all_ns"
+{ oc get storageclusters --all-namespaces -o yaml; } >> "${BASE_COLLECTION_PATH}/namespaces/all/storagesystem_yaml_all_ns"
```
Comment (jarrpa) [inline must-gather/collection-scripts/gather_namespaced_resources:121]: Part of this is duplicating something we're already doing above: https://github.com/red-hat-storage/ocs-operator/pull/1328/files#diff-22a4750fbab2abc8b4e080042968dc784aac21fe537db2e4089618d828efe866R91

Also, I don't remember why we're not just doing an `inspect` yet. Can you try just adding storageclusters and storagesystems to the `resources` array and see if it works?

**Review by jarrpa — CHANGES_REQUESTED** [review]

File: must-gather/collection-scripts/gather_namespaced_resources (hunk 7 above)
Comment (yati1998) [reply must-gather/collection-scripts/gather_namespaced_resources:121]: I used inspect but it failed to log the YAML files but will try it out.

Comment (jarrpa) [reply must-gather/collection-scripts/gather_namespaced_resources:121]: Hm, then the inspect problem isn't resolved yet, it sounds like. We should investigate this further. ...later. In any case, the main thing that matters is consistency. At present we don't need to worry about our resources outside the openshift-storage Namespace. As such, i suggest doing the StorageSystem get above where the current StorageCluster get is. There's no need to do a describe on them at this point if we aren't doing it for StorageClusters.

Comment (yati1998) [reply must-gather/collection-scripts/gather_namespaced_resources:121]: @umangachapagain suggested to add the describe command, but if it is not required, I shall remove it. Will do the changes as suggested by you and update the PR. Thanks @jarrpa

Comment (yati1998) [reply must-gather/collection-scripts/gather_namespaced_resources:121]: > Hm, then the inspect problem isn't resolved yet, it sounds like. We should investigate this further. ...later. In any case, the main thing that matters is consistency. At present we don't need to worry about our resources outside the openshift-storage Namespace. As such, i suggest doing the StorageSystem get above where the current StorageCluster get is. There's no need to do a describe on them at this point if we aren't doing it for StorageClusters.

Just a question regarding the above suggestion: 
1. Storagecluster is only for openshift namespace, but for storagesystem we are collecting logs for all the namespace right?
2. Do we need to collect logs for both `oc get storageclusters --all-namespaces -o yaml` and `oc get storagesystem --all-namespaces` cause I don't see the same for storagecluster.

**Review by jarrpa — CHANGES_REQUESTED** [review]
I think this can be simplified yet further, there's no need to treat storagesystems any differently than we do storageclusters.

>     1. Storagecluster is only for openshift namespace, but for storagesystem we are collecting logs for all the namespace right?

I assume you mean "openshift-storage", and no. We are currently only supporting both StorageClusters and StorageSystems in the openshift-storage namespace. I don't know why it was suggested to search all of them.

>     2. Do we need to collect logs for both `oc get storageclusters --all-namespaces -o yaml` and `oc get storagesystem --all-namespaces` cause I don't see the same for storagecluster.

No, you basically only need to do two things:

1. Duplicate this line for storagesystems, place it right below this: https://github.com/red-hat-storage/ocs-operator/pull/1328/files#diff-22a4750fbab2abc8b4e080042968dc784aac21fe537db2e4089618d828efe866R92
2. Add storagesystem to this array: https://github.com/red-hat-storage/ocs-operator/pull/1328/files#diff-22a4750fbab2abc8b4e080042968dc784aac21fe537db2e4089618d828efe866R51

File: must-gather/collection-scripts/gather_namespaced_resources (hunk 2 above)
[the remark this answers is not in the document]
Comment (yati1998) [reply must-gather/collection-scripts/gather_namespaced_resources:102]: Hey, can we keep this change? Cause adding the space at this line makes other changes.

**Review by agarwal-mudit — APPROVED** [review]
/lgtm
